NEWS.md

```diff
@@ -0,0 +1,232 @@
+2. When a user calls the `Close` (or `Reset`) method on a connection or stream,
+   go-libp2p will process the close event asynchronously (i.e., not block the
+   call to `Close`). Otherwise, a call to `Close` from within a connect event
+   handler would deadlock.
+3. Unless otherwise noted, events will be handled in parallel.
+
+What does this mean for end users? Well:
+
+1. Reference counting connections to a peer using connect/disconnect events
+   should "just work" and should never go negative.
+2. Under heavy connect/disconnect loads, connecting to new peers should be
+   faster (usually).
+   
+For those interested in the history of this issue, ...
+
+In the past, (dis)connect and stream open/close notifications have been a bit of
+a pain point. For a long time, they were fired off in parallel and one could, for
+example, process a disconnect notification before a connect notification (we had
+to support *negative* ref-counts in several places to account for this).
+
+After no end of trouble, we finally "fixed" this by synchronizing notification
+delivery. We still delivered notifications to all notifiees in parallel, we just
+processed the events in series.
+
+Unfortunately, under heavy connect/disconnect load, new connections could easily
+get stuck on open behind a queue of connect events all being handled in series.
+In theory, these events should have been handled quickly but in practice, it's
+very hard to avoid locks *entirely* (bitswap's event handlers were especially
+problematic).
+
+Worse, this serial delivery guarantee didn't actually provide us with an
+*in-order* delivery guarantee as it was still possible for a disconnect to
+happen before we even *started* to fire the connect event. The situation was
+slightly better than before because the events couldn't overlap but still far
+from optimal.
+
+However, this has all been resolved now. From now on, you'll never receive a
+disconnect event before a connect event.
+
+#### Conn.GetStreams Signature Change
+
+The signature of the `GetStreams` method on `go-libp2p-net.Conn` has changed from:
+
+```go
+GetStreams() ([]Stream, error)
+```
+
+To:
+
+```go
+GetStreams() []Stream
+```
+
+Listing the streams on an open connection should never involve IO or do anything
+that can fail so we removed this error to improve usability.
+
+### Libp2p Constructor
+
+If you're not already doing so, you should be using the `libp2p.New` constructor
+to make your libp2p nodes. This release brings quite a few new options to the
+libp2p constructor so if it hasn't been flexible enough for you in the past, I
+recommend that you try again. A simple example can be found in the
+[echo](https://github.com/libp2p/go-libp2p/tree/feat/refactor/examples/echo)
+example.
+
+Given this work and in an attempt to consolidate all of our configuration logic
+in one place, we've removed all default transports from go-libp2p-swarm.
+
+TL;DR: Please use the libp2p constructor.
+
+### Zombie Streams
```

Comment:
@bigs @mgoelzer 

This is a bit of an annoying libp2p quirk you should know about. We may eventually add some way to automatically garbage collect zombie streams but the lack of weak references in go makes this a bit difficult.